func.go

```diff
@@ -87,20 +87,49 @@ func (f *FuncValue) Apply(args ...interface{}) Slice {
 func (f *FuncValue) applyValue(args []reflect.Value) Slice {
 	argTypes := make([]reflect.Type, len(args))
 	for i, arg := range args {
+		if !arg.IsValid() {
+			if !checkNilAssignable(f.args[i]) {
+				// Untyped nil argument for type that cannot be nil.
+				typecheck.Panicf(2, "cannot use nil as type %s in argument to function", f.args[i])
+			}
+			argTypes[i] = f.args[i]
+			args[i] = reflect.Zero(f.args[i])
+			continue
+		}
 		argTypes[i] = arg.Type()
 	}
 	f.typecheck(argTypes...)
 	out := f.fn.Call(args)
 	return out[0].Interface().(Slice)
 }
 
+func checkNilAssignable(typ reflect.Type) bool {
+	switch typ.Kind() {
+	case reflect.Chan:
+	case reflect.Func:
+	case reflect.Interface:
+	case reflect.Map:
+	case reflect.Ptr:
+	case reflect.Slice:
+	default:
```

Comment:
Do we care about [`UnsafePointer`](https://go.googlesource.com/go/+/refs/tags/go1.13.3/src/reflect/value.go#1059)?